superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -589,19 +583,26 @@ function nvd3Vis(element, props) {
       const marginPad = Math.ceil(
         Math.min(maxWidth * (isExplore ? 0.01 : 0.03), MAX_MARGIN_PAD),
       );
-      const maxYAxisLabelWidth = getMaxLabelSize(svg, chart.yAxis2 ? 'nv-y1' : 'nv-y');
-      const maxXAxisLabelHeight = getMaxLabelSize(svg, 'nv-x');
-      chart.margin({ left: maxYAxisLabelWidth + marginPad });
-      if (yAxisLabel && yAxisLabel !== '') {
-        chart.margin({ left: maxYAxisLabelWidth + marginPad + 25 });
-      }
       // Hack to adjust margins to accommodate long axis tick labels.
       // - has to be done only after the chart has been rendered once
       // - measure the width or height of the labels
       // ---- (x axis labels are rotated 45 degrees so we use height),
       // - adjust margins based on these measures and render again
       const margins = chart.margin();
-      margins.bottom = 28;
+      if (chart.xAxis) {
+        margins.bottom = 28;
```

Comment:
just to understand -- we are modifying the reference to the chart margin object directly? should we instead copy it and call `chart.margin(newMargin)`? if it works I guess that's fine.

Reply:
Right. I understand how non-functional that is and didn't to go too deep in the refactor. I mostly just grouped all margin-related code in one section and made sure that logically would produce the same results. Hopefully we can get rid of this whole module and nvd3 fairly soon!